### PR TITLE
Interpret DOTNET_BUNDLE_EXTRACT_BASE_DIR without dir separator as relative path

### LINKED ISSUE #36804: Single-File: Linux: DOTNET_BUNDLE_EXTRACT_BASE_DIR: Fix interpretation of relative paths
On Linux, setting `DOTNET_BUNDLE_EXTRACT_BASE_DIR` to an absolute or relative path containing a `/` works OK whether the directory exists or not. However, paths without a `/` are not interpreted as a relative path. 

## Configuration
Target Platform: X64 Linux
Target Framework: .netcoreapp 3.1

## Repro Steps
`export DOTNET_BUNDLE_EXTRACT_BASE_DIR=asdf`
Run a hello-world single-file app.

## Expected outcome
A relative path `asdf` is created, and contents are extracted there. 

## Actual outcome
The run segfaults.

### PR DESCRIPTION
**Pull request by mateoatr**
Should fix #36804

### CODE REVIEW

File: src/installer/tests/Microsoft.NET.HostModel.Tests/AppHost.Bundle.Tests/BundleExtractToSpecificPath.cs [hunk 1]
```diff
@@ -57,6 +57,28 @@ private void Bundle_Extraction_To_Specific_Path_Succeeds()
...
+            // Run the bundled app (extract files to <path>)
+            Command.Create(singleFile)
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .EnvironmentVariable(BundleHelper.DotnetBundleExtractBaseEnvVariable, relativePath)
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdOutContaining("Hello World");
```
Comment (vitek-karas) [inline src/installer/tests/Microsoft.NET.HostModel.Tests/AppHost.Bundle.Tests/BundleExtractToSpecificPath.cs:79]: Nit: If it's not hard, you may also validate that it created the right folder and ran from there.

**Review by vitek-karas — APPROVED** [review]

**Review by vitek-karas — CHANGES_REQUESTED** [review]
I ran it just out of curiosity - and to be honest I don't understand why we're fixing it here...

The failure happens in `dir_utils_t::create_directory_tree` which ends up in an endless recursion. I think we should fix it there.

**Comment by vitek-karas** [comment]
Actually the root cause is here: https://github.com/dotnet/runtime/blob/master/src/installer/corehost/cli/hostmisc/utils.cpp#L143

This code doesn't handle relative paths well. Basically if I call `get_directory("name")` it returns `"name/"` and the caller (`create_direcotry_tree` in this case) expects to get back an empty string.

We should either make the utils work with relate paths (which might be pretty tricky) or we should turn the env. variable into an absolute path before we do anything else. So your fix sort of works in that regard, but it's still relatively fragile I think. I think a better solution would be to call `pal::realpath` - based on some docs it should work on just "name" input (and understand it as a "./name" basically).

**Comment by mateoatr** [comment]
The reason I did not use `realpath` originally was because it fails and returns null whenever the path does not exist (as in this case, since we're just about to create the dir), although we can do `realpath(".")` and then append the relative path...

I think the failures in Windows are due to cases when the path start with a volume, e.g., `C:\\foo` will become `.\\C:\\foo`, which fails.

**Comment by vitek-karas** [comment]
Good point about `realpath`. I think using the `realpath(".")` feels better - mainly because we can then declare a clear contract of the other methods - pass in absolute paths, always.

**Review by vitek-karas — APPROVED** [review]
Looks good

**Pull request merged**